mkosi/run.py

```diff
@@ -266,6 +266,19 @@ def run_with_apivfs(
         "--ro-bind", "/sys", state.root / "sys",
         "--bind", state.var_tmp, state.root / "var/tmp",
         "--die-with-parent",
+    ]
+
+    # If passwd or a related file exists in the root directory, bind mount it over the host files while we
+    # run the command, to make sure that the command we run uses user/group information from the root instead
+    # of from the host. If the file doesn't exist yet, mount over /dev/null instead.
+    for f in ("passwd", "group", "shadow", "gshadow"):
+        p = state.root / "etc" / f
+        if p.exists():
+            cmdline += ["--bind", p, f"/etc/{f}"]
+        else:
+            cmdline += ["--bind", "/dev/null", f"/etc/{f}"]
```

Comment:
This seems very sensible, but isn't there the potential for breakage here? At least in the Debian case I know that postinst scripts regularly look whether users exist and create them if they don't. I'm sure RPM postinst scriplets do the same. Assuming that the writes to e.g. `/etc/passwd` now go to `/dev/null`, won't the created user be missing in the image?

Reply:
If RPM or dpkg are writing to host /etc/passwd when installing in a chroot, something is seriously wrong. This fix is more about making sure that when installing files from an RPM or deb, the package managers use users/groups from the chroot instead of from the host. What could happen now is that instead of using the wrong uid for a user, the user simply won't be found anymore if we mount /dev/null over /etc/passwd. But I don't expect that to happen given the 2 stage package install process we use.